src/main/java/com/example/moviemanager/repository/model/Movie.java

```diff
@@ -74,14 +71,18 @@ public boolean equals(Object obj) {
         }
         Movie movie = (Movie) obj;
 
+        if(movie.id == null) {
```

Comment:
Why this check is needed? I think movie entity comparison should not happen before it was persisted